ZPR_SLOG_EMAIL -> ZPRT_EMAIL_SLOG, TSTMP_PRINT mezővel
Státusz: WIP. Dátum: 2026-10-16.

Döntés: a felhasználó kérésére kizárólag ez a diff frissül.
Kiindulás: a jelenlegi helyi munkapéldány, nem a git HEAD és nem az előző
patch alkalmazása utáni állapot. Ez a változat az előző diff helyébe lép.

A FELHASZNÁLÓ ÁLTAL LÉTREHOZOTTKÉNT JELZETT OBJEKTUMOK
- ZPRB_EMAIL_SLOG táblatípus.
- EZPRT_EMAIL_SLOG zárobjektum.
- ENQUEUE_EZPRT_EMAIL_SLOG és DEQUEUE_EZPRT_EMAIL_SLOG függvények.
Az objektumok létrehozását a felhasználó közölte; aktív definícióikat nem
olvastuk vissza SAP-ból.

ELŐFELTÉTELEK ÉS ELLENŐRZÉSEK
- ZPRB_EMAIL_SLOG: sortípus ZPRT_EMAIL_SLOG, táblakategória STANDARD TABLE.
  A meglévő kód APPEND, SORT és BINARY SEARCH műveleteire alkalmas legyen;
  az előző diff helyi táblatípusát ez a DDIC-táblatípus váltja fel.
- ZPRT_EMAIL_SLOG: TSTMP_PRINT CHAR 14, YYYYMMDDhhmmss formátumban, UTC
  szerint; nem kulcsmező. A diff MANDT + EXBILLDOCNO kulcsot feltételez.
  További mezők: OPBEL, EMAIL_ADDRESS, TSTMP_EMAIL_SENT, SENDING_STATUS,
  TSTMP_SENDING_STATUS, ERROR_MESSAGE.
- A küldő a TSTMP_PRINT mezőbe a korábbi logika szerinti Messenger-átadási
  időpontot írja. Ugyanazon küldésnél az érték változatlan maradjon.
  Meglévő soroknál is a valós eredeti időpontot kell megőrizni/átvezetni.
  A diff nem hozza létre a mezőt és nem tölti fel az adatokat.
- EZPRT_EMAIL_SLOG: alaptábla ZPRT_EMAIL_SLOG, zárkulcs MANDT + EXBILLDOCNO.
  A diff az ENQUEUE_EZPRT_EMAIL_SLOG / DEQUEUE_EZPRT_EMAIL_SLOG neveket és
  MODE_ZPRT_EMAIL_SLOG, MANDT, EXBILLDOCNO paramétereket használja.
  A generált interfészt, az aktív állapotot és a transzportot ellenőrizni
  kell. Az összes író ugyanazt a zárobjektumot használja.

A PATCH TARTALMA
- Főprogram, proxy, regisztrációs include és Unit-forrás: sortípusok,
  mezőtípusok, SELECT, UPDATE és INSERT átvezetése ZPRT_EMAIL_SLOG-ra.
- TY_EMAIL_TAB típusa ZPRB_EMAIL_SLOG lesz a régi ZPRB_SLOG_EMAIL helyett.
- A főprogram és a regisztrációs include összesen négy ENQUEUE/DEQUEUE
  hívása az új EZPRT_EMAIL_SLOG zárobjektum függvényeit használja;
  a módparaméter MODE_ZPRT_EMAIL_SLOG. A scope és a tranzakciókezelés marad.
- Az időbélyeg-segédmetódusok zprt_email_slog-tstmp_print típust használnak.
  A hétnapos N-státusz, az UPDATE időbélyegfeltétele, a regisztrációs
  ellenőrzések és az időbélyegre épülő tesztek logikája megmarad.
- A hiányzó GPART/VKONT mezők tesztadatai és assert-jei kikerülnek;
  OPBEL és EMAIL_ADDRESS megőrzésének ellenőrzése megmarad.

TOVÁBBI ELLENŐRIZENDŐ
- OPBEL az Excelben szerepel, megjegyzése opcionálisnak mondja. Ha az aktív
  táblában nincs, a hozzá tartozó tesztadat/assert is igazítandó.
- A küldőprogramok és a régi adatok átvezetése külön feladat. A regisztráció
  hívója az új sortípust és a kitöltött TSTMP_PRINT mezőt adja át.
- A korábbi SAP-kapcsolati hiba utáni leállás érvényben van. SAP-hívás,
  módosítás, fordítás, aktiválás, ATC és ABAP Unit-futtatás nem történt;
  az aktív SAP-állapot nincs ellenőrizve. A DDIC-előfeltételek után ezek és
  célzott regisztrációs/zárolási/mentési próbák szükségesek.
- Meglévő, e cserétől független QA-pont: a MESSAGE-/naplószövegek
  fordíthatósága ebben a diffben nincs rendezve.

diff --git "a/[03] projects/tomeges-nyomtatas-endre/04_ASSETS/abap/ZPR_MESSAGEQUEUE_QUERY.abap" "b/[03] projects/tomeges-nyomtatas-endre/04_ASSETS/abap/ZPR_MESSAGEQUEUE_QUERY.abap"
--- "a/[03] projects/tomeges-nyomtatas-endre/04_ASSETS/abap/ZPR_MESSAGEQUEUE_QUERY.abap"
+++ "b/[03] projects/tomeges-nyomtatas-endre/04_ASSETS/abap/ZPR_MESSAGEQUEUE_QUERY.abap"
@@ -21,18 +21,18 @@
 
 INCLUDE <icon>.
 
-TYPES ty_email_tab TYPE zprb_slog_email.
+TYPES ty_email_tab TYPE zprb_email_slog.
 TYPES ty_docno_tab TYPE STANDARD TABLE OF string WITH DEFAULT KEY.
 TYPES: BEGIN OF ty_result.
-         INCLUDE TYPE zpr_slog_email.
+         INCLUDE TYPE zprt_email_slog.
 TYPES:   save_state TYPE c LENGTH 1,
          save_text  TYPE c LENGTH 100,
        END OF ty_result.
 TYPES ty_result_tab TYPE STANDARD TABLE OF ty_result WITH DEFAULT KEY.
 
 CONSTANTS: gc_tcode TYPE sytcode VALUE 'ZPR_MQ_QUERY',
-           gc_missing TYPE zpr_slog_email-sending_status VALUE 'N',
+           gc_missing TYPE zprt_email_slog-sending_status VALUE 'N',
-           gc_missing_text TYPE zpr_slog_email-error_message
+           gc_missing_text TYPE zprt_email_slog-error_message
              VALUE 'Hét napon belül nem érkezett státusz a Messengertől.'.
 
 PARAMETERS p_lport TYPE prx_logical_port_name NO-DISPLAY.
@@ -53,9 +53,9 @@
       RETURNING VALUE(rv_stamp) TYPE timestamp.
     CLASS-METHODS format_stamp
       IMPORTING iv_stamp TYPE timestamp
-      RETURNING VALUE(rv_text) TYPE zpr_slog_email-tstmp_print.
+      RETURNING VALUE(rv_text) TYPE zprt_email_slog-tstmp_print.
     CLASS-METHODS expired
-      IMPORTING iv_print TYPE zpr_slog_email-tstmp_print
+      IMPORTING iv_print TYPE zprt_email_slog-tstmp_print
                 iv_now TYPE timestamp
       RETURNING VALUE(rv_expired) TYPE abap_bool.
 ENDCLASS.
@@ -160,7 +160,7 @@
 
 FORM run.
   DATA: lt_docnos TYPE ty_docno_tab,
-        ls_email TYPE zpr_slog_email,
+        ls_email TYPE zprt_email_slog,
         lv_docno TYPE string,
         lv_ok TYPE abap_bool,
         lv_error TYPE string,
@@ -168,7 +168,7 @@
 
   PERFORM add_log USING 'Messenger-státuszlekérdezés megkezdve.'.
   TRY.
-      SELECT * INTO TABLE gt_pending FROM zpr_slog_email
+      SELECT * INTO TABLE gt_pending FROM zprt_email_slog
         WHERE sending_status = space.
       IF gt_pending IS INITIAL.
         PERFORM add_log USING 'Nincs státusz nélküli e-mailnapló-bejegyzés.'.
@@ -218,8 +218,8 @@
 ENDFORM.
 
 FORM prepare_result.
-  DATA: ls_email TYPE zpr_slog_email,
+  DATA: ls_email TYPE zprt_email_slog,
-        ls_response TYPE zpr_slog_email,
+        ls_response TYPE zprt_email_slog,
         ls_result TYPE ty_result,
         lv_now TYPE timestamp,
         lv_expired TYPE abap_bool.
@@ -251,18 +251,18 @@
   ENDLOOP.
 ENDFORM.
 
-FORM unlock_email USING iv_docno TYPE zpr_slog_email-exbilldocno.
+FORM unlock_email USING iv_docno TYPE zprt_email_slog-exbilldocno.
-  CALL FUNCTION 'DEQUEUE_EZPR_SLOG_EMAIL'
+  CALL FUNCTION 'DEQUEUE_EZPRT_EMAIL_SLOG'
     EXPORTING
-      mode_zpr_slog_email = 'E'
+      mode_zprt_email_slog = 'E'
       mandt = sy-mandt
       exbilldocno = iv_docno
       _scope = '1'.
 ENDFORM.
 
 FORM save_result.
-  DATA: ls_original TYPE zpr_slog_email,
+  DATA: ls_original TYPE zprt_email_slog,
-        ls_current TYPE zpr_slog_email,
+        ls_current TYPE zprt_email_slog,
         lx_error TYPE REF TO cx_root,
         lv_text TYPE string,
         lv_docno TYPE string,
@@ -290,7 +290,7 @@
 
-    CALL FUNCTION 'ENQUEUE_EZPR_SLOG_EMAIL'
+    CALL FUNCTION 'ENQUEUE_EZPRT_EMAIL_SLOG'
       EXPORTING
-        mode_zpr_slog_email = 'E'
+        mode_zprt_email_slog = 'E'
         mandt = sy-mandt
         exbilldocno = <ls_result>-exbilldocno
         _scope = '1'
@@ -311,14 +311,14 @@
 
     TRY.
         CLEAR ls_current.
-        SELECT SINGLE * INTO ls_current FROM zpr_slog_email
+        SELECT SINGLE * INTO ls_current FROM zprt_email_slog
           WHERE exbilldocno = <ls_result>-exbilldocno.
         IF sy-subrc <> 0 OR ls_current <> ls_original.
           <ls_result>-save_state = 'C'.
           <ls_result>-save_text = 'A rekord megváltozott vagy törölték; új lekérdezés kell.'.
           lv_skipped = lv_skipped + 1.
         ELSE.
-          UPDATE zpr_slog_email
+          UPDATE zprt_email_slog
             SET sending_status = <ls_result>-sending_status
                 tstmp_sending_status = <ls_result>-tstmp_sending_status
                 error_message = <ls_result>-error_message
diff --git "a/[03] projects/tomeges-nyomtatas-endre/04_ASSETS/abap/ZPR_MQ_PROXY.abap" "b/[03] projects/tomeges-nyomtatas-endre/04_ASSETS/abap/ZPR_MQ_PROXY.abap"
--- "a/[03] projects/tomeges-nyomtatas-endre/04_ASSETS/abap/ZPR_MQ_PROXY.abap"
+++ "b/[03] projects/tomeges-nyomtatas-endre/04_ASSETS/abap/ZPR_MQ_PROXY.abap"
@@ -11,8 +11,8 @@
         ls_input TYPE zpr_query_messenger_request,
         ls_output TYPE zpr_query_messenger_response,
         ls_doc LIKE LINE OF ls_output-parameters-docs-doc,
-        ls_response TYPE zpr_slog_email,
+        ls_response TYPE zprt_email_slog,
-        ls_previous TYPE zpr_slog_email,
+        ls_previous TYPE zprt_email_slog,
         lv_docno TYPE string,
         lv_stamp_text TYPE string,
         lv_stamp TYPE timestamp,
diff --git "a/[03] projects/tomeges-nyomtatas-endre/04_ASSETS/abap/ZPR_MQ_REGISTER.abap" "b/[03] projects/tomeges-nyomtatas-endre/04_ASSETS/abap/ZPR_MQ_REGISTER.abap"
--- "a/[03] projects/tomeges-nyomtatas-endre/04_ASSETS/abap/ZPR_MQ_REGISTER.abap"
+++ "b/[03] projects/tomeges-nyomtatas-endre/04_ASSETS/abap/ZPR_MQ_REGISTER.abap"
@@ -3,11 +3,11 @@
 *& Does not commit or roll back the caller's LUW.
 *& After success the caller must close its LUW and release the lock.
 *&---------------------------------------------------------------------*
-FORM mq_register USING is_email TYPE zpr_slog_email
+FORM mq_register USING is_email TYPE zprt_email_slog
   CHANGING cv_ok TYPE abap_bool cv_error TYPE string
            cv_lock_held TYPE abap_bool.
-  DATA: ls_email TYPE zpr_slog_email,
+  DATA: ls_email TYPE zprt_email_slog,
-        ls_current TYPE zpr_slog_email,
+        ls_current TYPE zprt_email_slog,
         lv_stamp TYPE timestamp,
         lv_date TYPE d,
         lv_time TYPE t,
@@ -38,7 +38,7 @@
 
-  CALL FUNCTION 'ENQUEUE_EZPR_SLOG_EMAIL'
+  CALL FUNCTION 'ENQUEUE_EZPRT_EMAIL_SLOG'
     EXPORTING
-      mode_zpr_slog_email = 'E'
+      mode_zprt_email_slog = 'E'
       mandt = sy-mandt
       exbilldocno = is_email-exbilldocno
       _scope = '1'
@@ -53,7 +53,7 @@
   ENDIF.
   cv_lock_held = abap_true.
   TRY.
-      SELECT SINGLE * INTO ls_current FROM zpr_slog_email
+      SELECT SINGLE * INTO ls_current FROM zprt_email_slog
         WHERE exbilldocno = is_email-exbilldocno.
       IF sy-subrc = 0.
         IF ls_current-tstmp_print = is_email-tstmp_print.
@@ -64,7 +64,7 @@
       ELSE.
         ls_email = is_email.
         ls_email-mandt = sy-mandt.
-        INSERT zpr_slog_email FROM ls_email.
+        INSERT zprt_email_slog FROM ls_email.
         IF sy-subrc = 0.
           cv_ok = abap_true.
         ELSE.
@@ -79,12 +79,12 @@
   ENDIF.
 ENDFORM.
 
-FORM mq_register_unlock USING iv_docno TYPE zpr_slog_email-exbilldocno
+FORM mq_register_unlock USING iv_docno TYPE zprt_email_slog-exbilldocno
   CHANGING cv_lock_held TYPE abap_bool.
   IF cv_lock_held = abap_true.
-    CALL FUNCTION 'DEQUEUE_EZPR_SLOG_EMAIL'
+    CALL FUNCTION 'DEQUEUE_EZPRT_EMAIL_SLOG'
       EXPORTING
-        mode_zpr_slog_email = 'E'
+        mode_zprt_email_slog = 'E'
         mandt = sy-mandt
         exbilldocno = iv_docno
         _scope = '1'.
diff --git "a/[03] projects/tomeges-nyomtatas-endre/04_ASSETS/abap/test/ZPR_MQ_TESTS.abap" "b/[03] projects/tomeges-nyomtatas-endre/04_ASSETS/abap/test/ZPR_MQ_TESTS.abap"
--- "a/[03] projects/tomeges-nyomtatas-endre/04_ASSETS/abap/test/ZPR_MQ_TESTS.abap"
+++ "b/[03] projects/tomeges-nyomtatas-endre/04_ASSETS/abap/test/ZPR_MQ_TESTS.abap"
@@ -51,7 +51,7 @@
   ENDMETHOD.
 
   METHOD returned_status.
-    DATA: ls_email TYPE zpr_slog_email,
+    DATA: ls_email TYPE zprt_email_slog,
           ls_result TYPE ty_result.
     ls_email-exbilldocno = 'UNIT-STATUS'.
     APPEND ls_email TO gt_pending.
@@ -69,7 +69,7 @@
   ENDMETHOD.
 
   METHOD no_status_recent.
-    DATA: ls_email TYPE zpr_slog_email,
+    DATA: ls_email TYPE zprt_email_slog,
           lv_now TYPE timestamp.
     ls_email-exbilldocno = 'UNIT-RECENT'.
     GET TIME STAMP FIELD lv_now.
@@ -80,7 +80,7 @@
   ENDMETHOD.
 
   METHOD no_status_expired.
-    DATA: ls_email TYPE zpr_slog_email,
+    DATA: ls_email TYPE zprt_email_slog,
           ls_result TYPE ty_result.
     ls_email-exbilldocno = 'UNIT-EXPIRED'.
     ls_email-tstmp_print = '20000101000000'.
@@ -97,7 +97,7 @@
   METHOD char_timestamp_roundtrip.
     DATA: lv_stamp TYPE timestamp,
           lv_expected TYPE timestamp,
-          lv_text TYPE zpr_slog_email-tstmp_print.
+          lv_text TYPE zprt_email_slog-tstmp_print.
     lv_expected = '20261013120509'.
     lv_stamp = lcl_age=>parse_text( iv_text = '20261013120509' ).
     cl_abap_unit_assert=>assert_equals( act = lv_stamp exp = lv_expected ).
@@ -117,13 +117,11 @@
   ENDMETHOD.
 
   METHOD business_fields_preserved.
-    DATA: ls_email TYPE zpr_slog_email,
+    DATA: ls_email TYPE zprt_email_slog,
-          ls_response TYPE zpr_slog_email,
+          ls_response TYPE zprt_email_slog,
           ls_result TYPE ty_result.
     ls_email-exbilldocno = 'UNIT-BUSINESS'.
     ls_email-opbel = '000000000001'.
-    ls_email-gpart = '0000000002'.
-    ls_email-vkont = '000000000003'.
     ls_email-email_address = 'unit@example.invalid'.
     APPEND ls_email TO gt_pending.
     ls_response-exbilldocno = ls_email-exbilldocno.
@@ -133,8 +131,6 @@
     READ TABLE gt_result INDEX 1 INTO ls_result.
     cl_abap_unit_assert=>assert_subrc( exp = 0 ).
     cl_abap_unit_assert=>assert_equals( act = ls_result-opbel exp = ls_email-opbel ).
-    cl_abap_unit_assert=>assert_equals( act = ls_result-gpart exp = ls_email-gpart ).
-    cl_abap_unit_assert=>assert_equals( act = ls_result-vkont exp = ls_email-vkont ).
     cl_abap_unit_assert=>assert_equals(
       act = ls_result-email_address exp = ls_email-email_address ).
   ENDMETHOD.
